packages/eslint-plugin/src/rules/prefer-literal-enum-member.ts

```diff
@@ -59,16 +59,46 @@ export default createRule({
           return;
         }
 
+        // Allow bitwise operators when both operands are literals or refer to another member on this enum
         if (
           allowBitwiseExpressions &&
           node.initializer.type === AST_NODE_TYPES.BinaryExpression &&
           ['|', '&', '^', '<<', '>>', '>>>'].includes(
             node.initializer.operator,
           ) &&
-          node.initializer.left.type === AST_NODE_TYPES.Literal &&
-          node.initializer.right.type === AST_NODE_TYPES.Literal
+          node.parent.type === AST_NODE_TYPES.TSEnumDeclaration
         ) {
-          return;
+          const leftIsLiteral =
+            node.initializer.left.type === AST_NODE_TYPES.Literal;
+          const rightIsLiteral =
+            node.initializer.right.type === AST_NODE_TYPES.Literal;
+
+          // early branch, so gathering member identifiers only happens when needed
```

Comment:
[Proofreading] I like the previous comment, but this one doesn't really help with anything IMO. Early returns like this one are a common pattern.

```suggestion
```